### PR TITLE
Add retries to registration tasks

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: registration/tasks.py [hunk 1]
```diff
@@ -458,8 +459,8 @@ def create_subscription(contact):
             exc_info=True)
 
 
-@task()
-def jembi_post_json(registration_id):
+@task(bind=True, time_limit=10)
+def jembi_post_json(self, registration_id):
```
Comment (imsickofmaps) [inline registration/tasks.py]: `self` is for classes - take this out

File: registration/tasks.py [hunk 2]
```diff
@@ -475,9 +476,16 @@ def jembi_post_json(registration_id):
                 auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
                 verify=False
             )
-            return result.text
+            result.raise_for_status()
+        except HTTPError as e:
+            # retry message sending if in 500 range (3 default retries)
+            if 500 < e.response.status_code < 599:
+                raise self.retry(exc=e)
```
Comment (imsickofmaps) [inline registration/tasks.py]: should be `raise jembi_post_json.retry(exc=e)`

File: registration/tasks.py [hunk 3]
```diff
@@ -488,8 +496,8 @@ def jembi_post_json(registration_id):
             exc_info=True)
 
 
-@task()
-def jembi_post_xml(registration_id):
+@task(bind=True, time_limit=10)
+def jembi_post_xml(self, registration_id):
```
Comment (imsickofmaps) [inline registration/tasks.py]: as before

File: registration/tasks.py [hunk 4]
```diff
@@ -511,9 +519,16 @@ def jembi_post_xml(registration_id):
         try:
             result = requests.post(api_url, headers=headers, data=data,
                                    auth=auth, verify=False)
-            return result.text
+            result.raise_for_status()
+        except HTTPError as e:
+            # retry message sending if in 500 range (3 default retries)
+            if 500 < e.response.status_code < 599:
+                raise self.retry(exc=e)
```
Comment (imsickofmaps) [inline registration/tasks.py]: and here

File: registration/tasks.py [hunk 5]
```diff
@@ -553,23 +570,30 @@ def update_create_vumi_contact(registration_id, client=None):
...
+                    # Create new subscription for the contact
+                    subscription = create_subscription(contact)
+
+                    # Update the contact with subscription details
+                    updated_contact = update_contact_subscription(
+                        contact, subscription, client)
+
+                elif 500 < e.response.status_code < 599:
+                    # Retry task if 500 error
+                    raise self.retry(exc=e)
```
Comment (imsickofmaps) [inline registration/tasks.py]: and here

File: registration/tasks.py [hunk 6]
```diff
@@ -524,8 +539,8 @@ def jembi_post_xml(registration_id):
             exc_info=True)
 
 
-@task()
-def update_create_vumi_contact(registration_id, client=None):
+@task(bind=True, time_limit=10)
+def update_create_vumi_contact(self, registration_id, client=None):
```
Comment (imsickofmaps) [inline registration/tasks.py]: and here

**Comment by gsvr** [comment]
@imsickofmaps Re-review please.

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**